### PR TITLE
lays initial foundation for discord bot work

### PR DESCRIPTION
**Pull request by grishrl**
Copied over the code for loading in from files for all commands / events we want to use
refactored the web-api stuff I started with earlier today, 
we will have a folder for web-api, commands, and events
we can use this foundation to begin creating the discord bot

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,24 @@
 # bot
 Custom bot, built in discord.js, for use with NGS' Discord server.
+
+## Getting Started
+
+### Prerequisites
+- Node.js and NPM installation, you may need some extra peices for a few dependencies namely enmap check here for more (https://enmap.evie.codes/install)
```
Comment (carlosplusplus) [inline README.md]: 🎨- `pieces` is misspelled.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,24 @@
 # bot
 Custom bot, built in discord.js, for use with NGS' Discord server.
+
+## Getting Started
+
```
Comment (carlosplusplus) [inline README.md:5]: I'd love to use `nvm` here and lock the project to a specific version of `npm` and `node`.

Comment (grishrl) [reply README.md:5]: I have added node to the dependencies of the package - what will require having said node version and will cause npm to install that when running the install package; (however user will need node & npm (come together) in order to get a toe in the water)
From what I can tell nvm is just an environment manager to switch between node versions, am I missing something - I don't have nvm installed.

File: README.md (hunk 1 above)
Comment (grishrl) [reply README.md]: got it chief

File: README.md (hunk 2 above)
Comment (carlosplusplus) [reply README.md:5]: Yup, that's exactly right - `nvm` ensures that every single person running on this project is running the same version of `node` and `npm`. Without control there, you're going to run into situations where `npm install` might generate noise / garbage in `package.json` that varies from person to person, and now PRs become polluted.

Super easy to install - I'd pick the latest and greatest.

We can also consider using `yarn` over `npm`.

Comment (grishrl) [reply README.md:5]: Well I've been mulling it over today and I'm still a bit skeptical about the gains of doing this --
From what I can tell the nvm is a great tool especially for mac/linux users but from what i've gathered the nvmrc file doesn't work with the windows style nvm, it seems like it's a nice utility tool for folks who want to use nvm (sort of like a rb-env) but forcing it into the project doesn't seem like the right play to me.  
Further if we want to use yarn in leu or along side of npm then that should be fine I've seen lots of repos using both but I don't see any clear reason do one instead of the other
In short I don't fully see the need or gain from implementing another layer of complexity to the repo as lots of repos run with only node and npm, the package and package-lock handle the dependency tracking and it should only grow when we add new dependencies

Comment (carlosplusplus) [reply README.md:5]: All fine - if we run into any issues with package installation or `npm` specific changes to `package.json` due to specific versions (which is what this tries to address), we can revisit this later.

**Review by carlosplusplus — APPROVED** [review]
